zerver/lib/actions.py

```diff
@@ -4435,16 +4435,26 @@ def check_invite_limit(user: UserProfile, num_invitees: int) -> None:
 
         default_max = settings.INVITES_DEFAULT_REALM_DAILY_MAX
         newrealm_age = datetime.timedelta(days=settings.INVITES_NEW_REALM_DAYS)
-        if (user.realm.date_created > timezone_now() - newrealm_age
-                and user.realm.max_invites <= default_max):
-            new_realms = Realm.objects.filter(
-                date_created__gte=timezone_now() - newrealm_age,
-                _max_invites__lte=default_max,
-            ).all()
-            for days, count in settings.INVITES_NEW_REALM_LIMIT_DAYS:
-                recent_invites = estimate_recent_invites(new_realms, days=days)
-                if num_invitees + recent_invites > count:
-                    raise InvitationError(msg, [], sent_invitations=False)
+        if user.realm.date_created - timezone_now() > newrealm_age:
+            # If the user's realm is sufficiently old, then we don't
+            # apply invite limits.  This code is targeted at the specific
+            # scenario of somebody using a new realm to launch spam.
+            return
+
+        if user.realm.max_invites >= default_max:
```

Comment:
Logic changed here too. The `==` case is the most common one, so the behavior there is essential.